00_Base/src/util/command.executor.ts

```diff
@@ -298,6 +318,126 @@ export class CommandExecutor {
     );
   }
 
+  public async executeReserveNow(
+    reserveNow: ReserveNow,
+    countryCode: string,
+    partyId: string,
+  ): Promise<void> {
+    // Currently, it rejects the request if the evse_uid is missing
+    // When we have the solution to handle the optional evse_uid, we should change this
+    if (!reserveNow.evse_uid) {
+      throw new BadRequestError('Missing evse_uid');
+    }
+    const evse = await this.ocpiEvseEntityRepo.getOcpiEvseByEvseUid(
+      reserveNow.evse_uid!,
+    );
+    if (!evse) {
+      throw new NotFoundError('EVSE not found');
+    }
+    const ocpiLocation = await this.ocpiLocationRepo.readByKey(
+      reserveNow.location_id,
+    );
+    if (!ocpiLocation) {
+      throw new NotFoundError('Location not found');
+    }
+
```

Comment:
Do we need to validate that the EVSE belongs to the location that was found?

Reply:
This is a good idea. But I am not so clear about the relationship between ocpiEvse and ocpiLocation. Is this possible to do this validation? @yuvenus Do you know more about this?

Reply:
currently ocpiEvse and ocpiLocation don't hold relationships to each other because they are an extension of the ocpp-versions of those objects 😔 

however, there is already logic in `LocationsDatasource` to do the hard work to figure out the relationship. perhaps you can switch from using the repository to using the datasource and you should be able to use the method `getEvse()` which takes in a location id, station id (the first "half" of the evse uid") and the evse id (the second "half" of the evse uid).

there's a helper to split the evse uid into its two different parts, `EXTRACT_STATION_ID` and `EXTRACT_EVSE_ID` both of which take in the evse uid (it's complicated bc evse_uid is not an ocpp field and we have to make one up... 😬)

additionally, `getEvse()` will throw a NotFoundError if
* the location isn't found
* the evse doesn't belong to the location
* the evse isn't found

so i think you can shorten this ocpiEvse and ocpiLocation check into one check to location datasource's `getEvse()` 👀 (big win for the datasources, haha!)

so the code may look more like this:
```
public async executeReserveNow(
    reserveNow: ReserveNow,
    countryCode: string,
    partyId: string,
): Promise<void> {
  // Currently, it rejects the request if the evse_uid is missing
  // When we have the solution to handle the optional evse_uid, we should change this
  if (!reserveNow.evse_uid) {
    throw new BadRequestError('Missing evse_uid');
  }

  // below should already throw as per criteria listed previously
  const evse = await this.locationsDatasource.getEvse(
    reserveNow.location_id,
    EXTRACT_STATION_ID(reserveNow.evse_uid!),
    EXTRACT_EVSE_ID(reserveNow.evse_uid!)
  ); 

... the rest of the method

}
```

Reply:
Great! Thank you, Venus! I updated the code.